### PR TITLE
Remove GribWrapper auto-regularise.

### PR DESCRIPTION
**Pull request by bjlittle**
Remove the deprecated auto-regularise behaviour from `GribWrapper`

### CODE REVIEW

File: iris_grib/__init__.py [hunk 1]
```diff
@@ -199,7 +178,7 @@ class GribWrapper(object):
     provides alternative means of working with GRIB message instances.
 
     """
-    def __init__(self, grib_message, grib_fh=None, auto_regularise=True):
+    def __init__(self, grib_message, grib_fh=None):
         warn_deprecated('Deprecated at version 1.10')
```
Comment (marqh) [inline iris_grib/__init__.py]: i don't think this deprecation message is still vaild.  Is it?

**Comment by marqh** [comment]
this behaviour is GRIB2 specific, for a small number of GDTs and is minimally tested in iris currently

```
/scitools/iris/lib/iris/tests
> grep -iR auto_regularise *
unit/fileformats/grib/test_GribWrapper.py:        auto_regularise = False
unit/fileformats/grib/test_GribWrapper.py:            gw = GribWrapper(grib_message, grib_fh, auto_regularise)
unit/fileformats/grib/test_GribWrapper.py:            self.assertEqual(proxy.regularise, auto_regularise)
unit/fileformats/grib/test_GribWrapper.py:        auto_regularise = False
unit/fileformats/grib/test_GribWrapper.py:            gw = GribWrapper(grib_message, grib_fh, auto_regularise)
unit/fileformats/grib/test_GribWrapper.py:            self.assertEqual(proxy.regularise, auto_regularise)
unit/fileformats/grib/test_GribWrapper.py:        auto_regularise = False
unit/fileformats/grib/test_GribWrapper.py:            gw = GribWrapper(grib_message, grib_fh, auto_regularise)
unit/fileformats/grib/test_GribWrapper.py:            self.assertEqual(proxy.regularise, auto_regularise)
unit/fileformats/grib/test_GribWrapper.py:        auto_regularise = False
unit/fileformats/grib/test_GribWrapper.py:            gw = GribWrapper(grib_message, grib_fh, auto_regularise)
unit/fileformats/grib/test_GribWrapper.py:            self.assertEqual(proxy.regularise, auto_regularise)
unit/fileformats/grib/test_load_cubes.py:                    kw_args = {'auto_regularise': mock.sentinel.REGULARISE}
unit/fileformats/grib/test_load_cubes.py:        grib_generator = load_cubes(gribfile, auto_regularise=False)
Binary file unit/fileformats/grib/test_GribWrapper.pyc matches
Binary file unit/fileformats/grib/test_load_cubes.pyc matches

```

I think it is safe and sensible to remove

**Comment by bjlittle** [comment]
@marqh Okay, I agree :+1:

**Comment by bjlittle** [comment]
@marqh I'll wait for this to merge then, before rebasing #40

**Pull request merged**